### PR TITLE
Use the new CladeTime object in virus_clade_utils

### PR DESCRIPTION
**Pull request by elray1**
The CladeTime object encapsulates some of the boilerplate in the original version of this script (e.g., worrying about the config).

Additionally, we can tidy things up a bit not downloading the Nextstrain metadata before generating the counts.

### CODE REVIEW

File: get_covid_clade_counts.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+# /// script
+# requires-python = ">=3.11"
+# dependencies = [
+#   "click",
+#   "virus_clade_utils@git+https://github.com/reichlab/virus-clade-utils@bsweger/sequence-by-state-date/50"
+# ]
+# ///
+
+# ⬆️ Tell uv to use a the latest feature branch of virus-clade-utils until everything is merged
```
Comment (elray1) [inline get_covid_clade_counts.py:24]:
```suggestion
#   "virus_clade_utils@git+https://github.com/reichlab/virus-clade-utils"
# ]
# ///
```

File: get_covid_clade_counts.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
+"""
+Create a parquet file with clade counts by date and location and save it to an S3 bucket.
+
+This script wraps the virus-clade-utils package, which generates the clade counts using the
+GenBank-based Sars-CoV-2 sequence metadata from Nextstrain.
+https://github.com/reichlab/virus-clade-utils
```
Comment (bsweger) [inline get_covid_clade_counts.py:6]:
```suggestion
https://github.com/reichlab/cladetime
```

File: get_covid_clade_counts.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+To run the script manually:
+1. Install uv on your machine: https://docs.astral.sh/uv/getting-started/installation/
+2. From the root of this repo: uv run get_covid_clade_counts.py --as-of=YYYY-MM-DD
+"""
+
+# /// script
+# requires-python = ">=3.11"
+# dependencies = [
+#   "click",
+#   "virus_clade_utils@git+https://github.com/reichlab/virus-clade-utils"
```
Comment (bsweger) [inline get_covid_clade_counts.py:20]:
```suggestion
#   "cladetime@git+https://github.com/reichlab/cladetime"
```

File: get_covid_clade_counts.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+# ]
+# ///
+
+import click
+import os
+import logging
+
+from virus_clade_utils.cladetime import CladeTime  # type: ignore
```
Comment (bsweger) [inline get_covid_clade_counts.py:28]:
```suggestion
from cladetime import CladeTime  # type: ignore
```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,14 @@
 # get-covid-clade-counts
-Create virus clade count data and save to an S3 Bucket using functionality from virus-clade-utils
+Create virus clade count data in parquet format (eventually: and save to an S3 Bucket) using functionality from virus-clade-utils.
```
Comment (bsweger) [inline README.md:2]:
```suggestion
Create virus clade count data in parquet format (eventually: and save to an S3 Bucket) using functionality from cladetime.
```

File: get_covid_clade_counts.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
+"""
+Create a parquet file with clade counts by date and location and save it to an S3 bucket.
+
+This script wraps the virus-clade-utils package, which generates the clade counts using the
```
Comment (bsweger) [inline get_covid_clade_counts.py:4]:
```suggestion
This script wraps the cladetime package, which generates the clade counts using the
```

File: get_covid_clade_counts.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+handler = logging.StreamHandler()
+formatter = logging.Formatter(
+    "%(asctime)s -  %(levelname)s - %(message)s", datefmt="%m/%d/%Y %I:%M:%S %p"
+)
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+logger.setLevel(logging.INFO)
+
+
+data_dir = "./data"
```
Comment (bsweger) [inline get_covid_clade_counts.py:42]: If you use `pathlib` to do this, it should ["do the right thing" for people working on different systems](https://docs.python.org/3/library/pathlib.html#pathlib.Path) (Windows, for example). And I learned recently that `Path` has a `mkdir` method, so you can do it all without `os` 🎉 

```
from pathlib import Path

data_dir = Path("./data")
data_dir.mkdir(exist_ok=True)
```

File: get_covid_clade_counts.py [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    # get Polars LazyFrame to Nextstrain sequence metadata
+    lf_metadata = ct.sequence_metadata
+
+    logger.info("filter_covid_genome_metadata")
+    lf_metadata_filtered = filter_covid_genome_metadata(lf_metadata)
+    logger.info("get_clade_counts")
+    counts = get_clade_counts(lf_metadata_filtered)
+
+    output_file = f"data/{as_of}_covid_clade_counts.parquet"
+    logger.info("collect")
```
Comment (bsweger) [inline get_covid_clade_counts.py:72]:
```suggestion
    logger.info("collecting clade counts")
```

Comment (bsweger) [reply get_covid_clade_counts.py:72]: In case people wonder what we're collecting

File: get_covid_clade_counts.py [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    cc = counts.collect()
```
Comment (bsweger) [inline get_covid_clade_counts.py:73]: Even though the Polars docs are pretty clear that it's not a stable feature, it's probably worth using the `streaming=True` option for `collect()`, to ensure that Polars handles any potential larger-than-memory data:
https://docs.pola.rs/api/python/stable/reference/lazyframe/api/polars.LazyFrame.collect.html

**Review by bsweger — COMMENTED** [review]
This is great--thanks for being patient with the rapidly evolving `cladetime` code base!

I never know whether to make comments or actual suggestions, so I've annoyingly done a little of both.

Changes required for the package name change are suggestions.

There are other minor notes, but those aren't critical if the intent is to run this script via a GitHub action.

File: get_covid_clade_counts.py (hunk 7 above)
Comment (elray1) [reply get_covid_clade_counts.py]: done

File: get_covid_clade_counts.py (hunk 9 above)
Comment (elray1) [reply get_covid_clade_counts.py]: done

**Review by bsweger — APPROVED** [review]

**Pull request merged**